test/test_mahalanobis_mixin.py

```diff
@@ -0,0 +1,203 @@
+import numpy as np
+from numpy.testing import assert_array_almost_equal
+from sklearn import clone
+from sklearn.datasets import load_iris
+from sklearn.utils import check_random_state, shuffle
+
+from metric_learn import (Constraints, ITML, LSML, MMC, SDML, Covariance, LFDA,
+                          LMNN, MLKR, NCA, RCA)
+from metric_learn.constraints import wrap_pairs
+from functools import partial
+
+
+def build_data():
+  RNG = check_random_state(0)
+  dataset = load_iris()
+  X, y = shuffle(dataset.data, dataset.target, random_state=RNG)
+  num_constraints = 20
+  constraints = Constraints.random_subset(y)
+  pairs = constraints.positive_negative_pairs(num_constraints,
+                                              same_length=True,
+                                              random_state=RNG)
+  return X, pairs
+
+
+def build_pairs():
+  # test that you can do cross validation on tuples of points with
+  #  a WeaklySupervisedMetricLearner
+  X, pairs = build_data()
+  pairs, y = wrap_pairs(X, pairs)
+  pairs, y = shuffle(pairs, y)
+  return pairs, y
+
+
+def build_quadruplets():
+  # test that you can do cross validation on a tuples of points with
+  #  a WeaklySupervisedMetricLearner
+  X, pairs = build_data()
+  c = np.column_stack(pairs)
+  quadruplets = X[c]
+  quadruplets = shuffle(quadruplets)
+  return quadruplets, None
+
+
+list_estimators = [(Covariance(), build_data),
+                   (ITML(), build_pairs),
+                   (LFDA(), partial(load_iris, return_X_y=True)),
+                   (LMNN(), partial(load_iris, return_X_y=True)),
+                   (LSML(), build_quadruplets),
+                   (MLKR(), partial(load_iris, return_X_y=True)),
+                   (MMC(), build_pairs),
+                   (NCA(), partial(load_iris, return_X_y=True)),
+                   (RCA(), partial(load_iris, return_X_y=True)),
+                   (SDML(), build_pairs)
+                   ]
+
+ids_estimators = ['covariance',
+                  'itml',
+                  'lfda',
+                  'lmnn',
+                  'lsml',
+                  'mlkr',
+                  'mmc',
+                  'nca',
+                  'rca',
+                  'sdml',
+                  ]
+
+
+@pytest.mark.parametrize('estimator, build_dataset', list_estimators,
+                         ids=ids_estimators)
+def test_score_pairs_pairwise(estimator, build_dataset):
+  # Computing pairwise scores should return an euclidean distance matrix.
+  inputs, labels = build_dataset()
+  X, _ = load_iris(return_X_y=True)
+  n_samples = 20
+  X = X[:n_samples]
+  model = clone(estimator)
+  model.fit(inputs, labels)
+
+  pairwise = model.score_pairs(np.array(list(product(X, X))))\
+      .reshape(n_samples, n_samples)
+
+  check_is_distance_matrix(pairwise)
+
+  # a necessary condition for euclidean distances matrix: (see
+  # https://en.wikipedia.org/wiki/Euclidean_distance_matrix)
+  assert np.linalg.matrix_rank(pairwise**2) <= min(X.shape) + 2
```

Comment:
this should be square root instead of square?

Reply:
same as above: `score_pairs` returns true euclidean distances, so I guess it is OK here too